test/concepts/test_concept_scheduler.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) Lucian Radu Teodorescu
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <catch2/catch.hpp>
+#include <execution.hpp>
+
+namespace ex = std::execution;
+
+struct my_scheduler {
+  struct my_sender {
+    template <template <class...> class Tuple, template <class...> class Variant>
+    using value_types = Variant<Tuple<>>;
+    template <template <class...> class Variant>
+    using error_types = Variant<std::exception_ptr>;
+    static constexpr bool sends_done = true;
+
+    template <typename CPO>
+    friend my_scheduler tag_invoke(ex::get_completion_scheduler_t<CPO>, my_sender) {
+      return {};
+    }
+  };
+
+  friend my_sender tag_invoke(ex::schedule_t, my_scheduler) { return {}; }
+
+  friend bool operator==(my_scheduler, my_scheduler) noexcept { return true; }
+  friend bool operator!=(my_scheduler, my_scheduler) noexcept { return false; }
+};
+
+TEST_CASE("type with schedule CPO models scheduler", "[concepts][scheduler]") {
+  REQUIRE(ex::scheduler<my_scheduler>);
+  // NOLINTNEXTLINE
+  auto snd = ex::schedule(my_scheduler{});
+  REQUIRE(ex::sender<decltype(snd)>);
```

Comment:
I would rather combine that into a single line
```suggestion
  REQUIRE(ex::sender<decltype(ex::schedule(my_scheduler{}))>);
```